openpgp-dsm/src/lib.rs

```diff
@@ -777,52 +811,59 @@ pub fn generate_key(
         cert = cert.insert_packets(vec![Packet::from(uid), uid_sig.into()])?;
     }
 
-    info!("key generation: sign encryption subkey");
-    {
-        let (subkey, flags) = (&encryption_subkey, &encryption_subkey_flags);
-        let pk: Key<PublicParts, SubordinateRole> = subkey
-            .sequoia_key.as_ref().context("unloaded subkey")?.clone().into();
-
-        let builder = SignatureBuilder::new(SignatureType::SubkeyBinding)
-            .set_key_validity_period(validity_period)?
-            .set_hash_algo(hash_algo)
-            .set_signature_creation_time(pk.creation_time())?
-            .set_key_flags(flags.clone())?;
-
-        let signature = pk.bind(&mut prim_signer, &cert, builder)?;
-        cert = cert.insert_packets(vec![Packet::from(pk), signature.into()])?;
+    if !encryption_subkeys.is_empty() {
+        info!("key generation: sign encryption subkey");
+        {
+            let (subkey, flags) = (&encryption_subkeys[0], &encryption_subkey_flags);
```

Comment:
Why only the first key?

Reply:
While the code is written to be able to create multiple subkeys (support in future), as per discussion with @zugzwang, for now we have decided to support only one signing subkey and one encryption subkey.

Reply:
correct, but this is not properly sanitized; in this case we should add some validation at the beginning of this function asserting that we are in one of the two cases: `CS,EtEr` or `C,S,EtEr`, and error with unsupported if this is not the case. Also, the less indexing the better (perhaps iterators, or `.first`, etc).

Reply:
ok i see zugzwang already commented to that effect: https://github.com/fortanix/sq-dsm/pull/77#discussion_r1384635908

Reply:
Given that we're only supporting one key for now, do you want to make this more clear in the code (e.g., use an `Option<PublicKey>` instead of a vector? (Or are you using a vector in anticipation of when we'll support more keys?)

Reply:
Vector is added in anticipation of when we will support multiple subkeys.
Updating the code as per other comments to make it more clear.

Reply:
Added validation of argument at start of function.

Reply:
resolving to reduce clutter